airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -0,0 +1,139 @@
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from datetime import date
+from dateutil.relativedelta import *
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+
+from .google_ads import GoogleAds
+from .utils import Utils
+
+
+def chunk_date_range(start_date: str, conversion_window: Optional[int]) -> Iterable[Mapping[str, any]]:
+    """
+    Returns a list of the beginning and ending timetsamps of each month between the start date and now.
+    The return value is a list of dicts {'date': str} which can be used directly with the Slack API
+    """
+    intervals = []
+
+    # As in to return some state when state in abnormal
+    if start_date > date.today().isoformat():
+        return [{"date": start_date}]
+
+    # applying conversion windoe
+    start_date = date.fromisoformat(
+        start_date) - relativedelta(days=conversion_window)
+    yesterday = date.today() - relativedelta(days=1)
+
+    # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
+    while start_date < yesterday:
+        start = start_date
+        intervals.append({"date": start.isoformat()})
+        start_date = min(yesterday, start_date + relativedelta(months=1))
+
+    return intervals
+
+
+class GoogleAdsStream(Stream, ABC):
+    def __init__(self, config):
+        self.config = config
+        self.google_ads_client = GoogleAds(**config)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
if this is that simple then I'd say the `next_page_token` method is not needed, we can just directly call `response.next_page_token`